modules/gapi/include/opencv2/gapi/garray.hpp

```diff
@@ -73,7 +82,23 @@ namespace detail
 
         void setConstructFcn(ConstructVec &&cv);  // Store T-aware constructor
 
+        template <typename T>
+        void specifyType();                       // Store type of initial GArray<T>
+
         std::shared_ptr<GOrigin> m_priv;
+        std::shared_ptr<TypeHintBase> m_hint;
```

Comment:
can it be `unique_ptr`?

Reply:
And the name should reflect it's purpose (I mean should have `type` related words in it)

Reply:
> can it be unique_ptr?

No, since GArrayU objects are copyable, and it is easier to share a type hint in this case.